src/ast.ts

```diff
@@ -299,34 +299,11 @@ function augmentFieldNodeTree(
  * @param compilations
  */
 function joinShouldIncludeCompilations(...compilations: string[]) {
-  // remove empty strings
   let filteredCompilations = compilations.filter(it => it);
-
-  // FIXME(boopathi): Maybe, find a better fix?
-  //
-  // remove "true" since we are joining with '&&' as `true && X` = `X`
-  // This prevents an explosion of `&& true` which could break
-  // V8's internal size limit for string.
-  //
-  // Note: the `true` appears if a field does not have a skip/include directive
-  // So, the more nested the query is, the more of unnecessary `&& true`
-  // we get.
-  //
-  // Failing to do this results in [RangeError: invalid array length]
-  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Invalid_array_length
-  filteredCompilations = filteredCompilations.filter(
-    it => it.trim() !== "true"
+  filteredCompilations = ([] as string[]).concat(
+    ...filteredCompilations.map(e => e.split(" && "))
   );
-
-  /**
-   * If we have filtered out everything in the list, it means, we
-   * called this with a join(true, true, etc...) - returning empty string
-   * is wrong, so we return a single true
-   */
-  if (filteredCompilations.length < 1) {
-    return "true";
-  }
-
+  filteredCompilations = Array.from(new Set(filteredCompilations));
```

Comment:
Would there be any helpful line comment to add for the future reader, why `Array.from(new Set(..))` is needed?